### ISSUE TITLE
Loss and ave loss NaN

### ISSUE THREAD

**Issue by guillepm97** [thtrieu/darkflow#1059]
Hi,
I´m trying to train a model that is based on 7000 images. I´ve tried training my model with a learning rate of 0,1 and 0,3 with trainer adam and batch 64 (and GPU) but the loss and ave loss get Nan in early steps. Now I'm trying to do the same but with a learning rate of 0,01. It trains such more steps and epochs, but when the loss and ave loss is near to 0.10-0.09, these values change to Nan again. Do you know if it is related to some any of this parameters or it is because my model could not train more? The last option should not be because it should be training until all epochs are finished.
Thank you for your time.

**Comment by Ridhwanluthra** [comment]
its probably an error in the calculation of the loss. did you make any changes or used darkflow out of the box? maybe try a debugger to catch where you get the first NAN and share that

**Comment by S0soo** [comment]
use your own dataset? If the loss is nan at first, maybe you need to check your labels. If you are training for a while, you need to reduce the learning rate. The premise is that you have not made any changes to darkflow.

**Comment by Tanmay-Kulkarni101** [comment]
Another reason, why a person may encounter Nan, is when the learning rate is too high, and the loss rather than converging , tends to diverge. Did you try making use of a different optimizer, especially one that makes the learning rate decay over time ?

You can check out this link to get an idea of what I am talking about -
https://ruder.io/optimizing-gradient-descent/
